### ISSUE TITLE
inline iframes require title attributes - a11y

### ISSUE THREAD

**Issue by jwinkworth** [ampproject/amphtml#29874]
## What's the issue?

AMP components that create inline iframes do not have a title attribute. This is against [WCAG guidelines](https://www.w3.org/TR/WCAG20-TECHS/H64.html) and fails [Lighthouse audits](https://web.dev/frame-title/?utm_source=lighthouse&utm_medium=devtools).

To be clear, this is not an issue with the &lt;amp-iframe&gt; component, but with other components that under the hood create an inline iframe.

Similar bugs appears to have been logged multiple times in the past here, with some reports closed/merged claiming a fix.

- #12617 - appears to be the PR for this very issue, but I am unclear of it's status (last updated 2 years ago)
- #17308 - this issue is specific to amp-iframe, but the issue is larger than just that component

## How do we reproduce the issue?

1. Visit http://content-solutions.s3.ca-central-1.amazonaws.com/amp-inline-iframe-bug/index.html
2. Inspect either inline iframe (youtube/brightcove). Both amp tags have a title attribute defined

### Expected result
The inline iframe created by AMP should have the title attribute attached.

### Actual result
The inline iframe created by AMP does not have a title attribute attached.

## What browsers are affected?

All browsers, all OSes

## Which AMP version is affected?

2007302351001

**Comment by caroqliu** [comment]
@realPrimoh Will be working on this - for some reason you don't come up in the assignees UI. 😅

**Comment by westonruter** [comment]
Just to note that `amp-iframe` _does_ propagate the `title` attribute to the `iframe` (as of #27037), as this attribute is among the `ATTRIBUTES_TO_PROPAGATE`:

https://github.com/ampproject/amphtml/blob/7785e940fbac13bb0314d78a2a270d6219c84b66/extensions/amp-iframe/0.1/amp-iframe.js#L51-L62

But `amp-youtube` (and others probably) [does not](https://github.com/ampproject/amphtml/blob/7785e940fbac13bb0314d78a2a270d6219c84b66/extensions/amp-youtube/0.1/amp-youtube.js#L264-L309). I made this example to show the difference between `amp-youtube` and `amp-iframe`, where the former does not copy the `title` attribute but the latter does: https://amp-iframe-title-attribute-iframe-copying.glitch.me/

For the `amp-youtube`, there is a Lighthouse accessibility audit failure:

> ![image](https://user-images.githubusercontent.com/134745/96917908-50cc9c00-145e-11eb-8835-282153520a14.png)

**Comment by realPrimoh** [comment]
@westonruter Thanks Weston, this is good to know. It seems to be mostly an internal iframe issue where the iframe is created within the AMP element to display external sites, for things like amp-youtube, amp-instagram, amp-facebook, etc.

**Comment by westonruter** [comment]
Yes, I think the main issue is that the same `ATTRIBUTES_TO_PROPAGATE` logic needs to be adapted from `amp-iframe` into the other AMP components that also create `iframe`.

**Comment by realPrimoh** [comment]
This has been fixed and merged as of this week: https://github.com/ampproject/amphtml/pull/30834.

**Comment by jwinkworth** [comment]
I've noticed that while this general iframe fix has gone live, the two test examples above aren't fixed.

The Brightcove video was missed in the MR that fixed this, and the YouTube video will always have the title of "AMP YouTube Video" instead of allowing for a title to be provided by the author.

YouTube
https://github.com/ampproject/amphtml/blob/6cba3aa84378c86d1f8ed325231ff68ce22b1297/extensions/amp-youtube/0.1/amp-youtube.js#L264

Brightcove
https://github.com/ampproject/amphtml/blob/6cba3aa84378c86d1f8ed325231ff68ce22b1297/extensions/amp-brightcove/0.1/amp-brightcove.js#L127-L130
